src/pages/[redirect]/index.tsx

```diff
@@ -0,0 +1,96 @@
+import fetchOriginalUrl from '@/utils/fetchOriginalUrl';
+import RedirectIcon from '../../../public/assets/icons/redirect';
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import Head from 'next/head';
+import { TINY_SITE } from '@/constants/url';
+
+const Redirect = () => {
+    const router = useRouter();
+    const { redirect: shortUrlCode } = router.query as { redirect: string };
+    const [originalUrl, setOriginalUrl] = useState('');
+    const [timer, setTimer] = useState(5);
+    const [isPremiumUser, setIsPremiumUser] = useState(false);
+    const [showTooltip, setShowTooltip] = useState(false);
+
+    useEffect(() => {
+        setIsPremiumUser(false);
+
+        if (shortUrlCode) {
```

Comment:
can we check if it has length as its a string so that it doesn't be empty string

Reply:
If the `url `is null, the else block is executed, and the code navigates to the home page using router.push('/').